src/transport/mod.rs

```diff
@@ -19,11 +20,12 @@ pub use self::{
 
 fn set_nodelay_or_warn(socket: &::tokio::net::TcpStream) {
     if let Err(e) = socket.set_nodelay(true) {
-        tracing::warn!(
-            "could not set TCP_NODELAY on {:?}/{:?}: {}",
-            socket.local_addr(),
-            socket.peer_addr(),
-            e
-        );
+        tracing::warn!("failed to set nodelay: {}", e);
+    }
+}
+
+pub fn set_keepalive_or_warn(tcp: &mut TcpStream, ka: Option<Duration>) {
+    if let Err(e) = tcp.set_keepalive(ka) {
+        tracing::warn!("failed to set keepalive: {}", e);
```

Comment:
TIOLI: record the error as a structured field:
```suggestion
        tracing::warn!(error = %e, "failed to set keepalive");
```